usl_pipeline/study_area_uploader/study_area_uploader/transformers/study_area_transformers.py

```diff
@@ -0,0 +1,148 @@
+        )
+
+
+def transform_study_area_files(
+    output_dir_path: pathlib.Path | str,
+    elevation_file_path: pathlib.Path | str,
+    sub_area_boundaries_shape_file_path: Optional[pathlib.Path | str] = None,
+    buildings_shape_file_path: Optional[pathlib.Path | str] = None,
+    green_areas_shape_file_path: Optional[pathlib.Path | str] = None,
+    soil_classes_shape_file_path: Optional[pathlib.Path | str] = None,
+    soil_classes_shape_file_property: Optional[str] = None,
+    elevation_output_file_name: str = file_names.ELEVATION_TIF,
+    boundaries_output_file_name: str = file_names.BOUNDARIES_TXT,
+    buildings_output_file_name: str = file_names.BUILDINGS_TXT,
+    green_areas_output_file_name: str = file_names.GREEN_AREAS_TXT,
+    soil_classes_output_file_name: str = file_names.SOIL_CLASSES_TXT,
+) -> pathlib.Path:
+    """Function prepares study area input files in a format expected by the pipeline.
+
+    Most of the files (except elevation one) are optional. If sub-area boundaries file
+    is defined all input files are cropped to the bounding box calculated for the
+    boundaries. Output files are created in the output directory (except for the case of
+    elevation file with no need to crop it, in this case the original file should be
+    used).
+
+    Args:
+        output_dir_path: Directory for output files.
+        elevation_file_path: Path to input elevation data.
+        sub_area_boundaries_shape_file_path: Optional shape-file with boundaries that
+            is used to crop study area.
+        buildings_shape_file_path: Optional shape-file with buildings.
+        green_areas_shape_file_path: Optional shape-file with green areas.
+        soil_classes_shape_file_path: Optional shape-file with soil classes
+        soil_classes_shape_file_property: Optional property name in soil classes shape
+            file that defines soil class values.
+        elevation_output_file_name: Default elevation output file name.
+        boundaries_output_file_name: Default boundaries output file name.
+        buildings_output_file_name: Default buildings output file name.
+        green_areas_output_file_name: Default green areas output file name.
+        soil_classes_output_file_name: Default soil classes output file name.
+
+    Returns:
+        Output file path to elevation data (can point to input file path in case no
+        changes are required for it).
+    """
+    with pathlib.Path(elevation_file_path).open("rb") as input_file:
+        elevation_header = elevation_readers.read_from_geotiff(
+            input_file, header_only=True
+        ).header
+    crs = elevation_header.crs.to_string()
+
+    output_dir = pathlib.Path(output_dir_path)
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
+
+    sub_area_bbox: Optional[geo_data.BoundingBox] = None
+    if sub_area_boundaries_shape_file_path is not None:
+        print("Preparing Boundaries...")
```

Comment:
Looks like a debug line here & a few spots below?
Or did you intend for this to be user output when run via a CLI?
I think it's usually better to use logging facilities `logging.info("Preparing Boundaries...")` from library functions, as the binary's `main` can configure where the logging output should go (stdout, a file, etc.)

Reply:
Done